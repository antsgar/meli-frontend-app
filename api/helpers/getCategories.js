const getCategories = ({ path_from_root }) => {
    return path_from_root.map(({ name }) => name);
};

module.exports = getCategories;
