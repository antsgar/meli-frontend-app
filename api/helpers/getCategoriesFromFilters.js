const getCategories = require("./getCategories");

const getCategoriesFromFilters = (filters) => {
    const categoryFilter = filters.find(
        ({ id, values }) => id === "category" && values.length > 0
    );
    return categoryFilter ? getCategories(categoryFilter.values[0]) : [];
};

module.exports = getCategoriesFromFilters;
