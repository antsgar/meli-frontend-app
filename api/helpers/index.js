const getCategories = require("./getCategories");
const getCategoriesFromFilters = require("./getCategoriesFromFilters");
const getItem = require("./getItem");
const getItems = require("./getItems");

module.exports = {
    getCategories,
    getCategoriesFromFilters,
    getItem,
    getItems,
};
