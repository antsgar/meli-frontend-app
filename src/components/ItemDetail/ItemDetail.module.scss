@import "../../styles/mixins";
@import "../../styles/vars";

.container {
    @include gridWrapper(12);

    flex: 1;
    background-color: $background-color-default;
    padding-bottom: $spacing-large;

    @include desktop {
        padding-left: $spacing-large;
        padding-right: $spacing-large;
    }
}

.content {
    @include gridContent(2, 12);
}

.card {
    background-color: white;
    border-radius: 4px;
}

.cardContent {
    @include gridWrapper(10);
}

.pictureContainer {
    @include gridContent(1, 11);
    padding: $spacing-medium;
    margin: auto;

    @include desktop {
        @include gridContent(1, 8);
        width: 100%;
        max-width: 680px;
        padding: $spacing-large 0 0 $spacing-large;
    }
}

.picture {
    height: 100%;
    width: 100%;
    object-fit: contain;
    border-radius: 4px;

    @include desktop {
        max-width: 680px;
        max-height: 680px;
    }
}

.productInfoContainer {
    @include gridContent(1, 11);
    padding: $spacing-medium;

    @include desktop {
        @include gridContent(8, 11);
        padding: $spacing-large $spacing-large 0 0;
    }
}

.condition {
    font-size: 14px;
    color: $text-color-dark;
}

.title {
    font-size: 24px;
    color: $text-color-dark;
    font-weight: bold;
    margin-top: $spacing-medium;
}

.price {
    font-size: 46px;
    color: $text-color-dark;
    margin-top: $spacing-large;
}

.button {
    background-color: $button-color-default;
    color: white;
    font-size: 18px;
    width: 100%;
    padding: $spacing-medium;
    margin-top: $spacing-large;
    border-radius: 4px;

    &:hover {
        opacity: 0.9;
        cursor: pointer;
    }
}

.descriptionContainer {
    @include gridContent(1, 11);
    padding: $spacing-large;

    @include desktop {
        @include gridContent(1, 7);
    }
}

.descriptionHeader {
    font-size: 28px;
    color: $text-color-dark;
}

.description {
    font-size: 16px;
    color: $text-color-default;
    margin-top: $spacing-large;
    overflow: hidden;
    overflow-wrap: break-word;
    white-space: break-spaces;
}
