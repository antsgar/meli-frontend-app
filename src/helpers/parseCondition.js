const parseCondition = (condition) => {
    switch (condition) {
        case "new":
            return "Nuevo";
        case "used":
            return "Usado";
        default:
            return "Nuevo";
    }
};

export default parseCondition;
