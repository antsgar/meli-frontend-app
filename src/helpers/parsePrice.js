const parsePrice = (price) => {
    return new Intl.NumberFormat("es-AR", {
        style: "currency",
        currency: price.currency,
        minimumFractionDigits: 0,
    }).format(Math.round(price.amount));
};

export default parsePrice;
