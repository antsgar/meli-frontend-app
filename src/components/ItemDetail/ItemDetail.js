import { Breadcrumb, SpinnerSupport } from "..";
import { parseCondition, parsePrice } from "../../helpers";
import styles from "./ItemDetail.module.scss";

const ItemDetail = ({ categories, item }) => {
    const { picture, condition, quantity, price, title, description } = item;

    return (
        <div className={styles.container} data-test="item-detail">
            <div className={styles.content}>
                <SpinnerSupport>
                    <Breadcrumb breadcrumbItems={categories} />
                    <div className={styles.card}>
                        <div className={styles.cardContent}>
                            <div className={styles.pictureContainer}>
                                <img
                                    className={styles.picture}
                                    src={picture}
                                    alt="Product picture"
                                ></img>
                            </div>
                            <div className={styles.productInfoContainer}>
                                <div className={styles.condition}>
                                    {parseCondition(condition)} - {quantity} vendidos
                                </div>
                                <div className={styles.title}>{title}</div>
                                <div className={styles.price}>
                                    {parsePrice(price)}
                                </div>
                                <button type="button" className={styles.button}>
                                    Comprar
                                </button>
                            </div>
                            <div className={styles.descriptionContainer}>
                                <div className={styles.descriptionHeader}>
                                    Descripci√≥n
                                </div>
                                <div className={styles.description}>
                                    {description}
                                </div>
                            </div>
                        </div>
                    </div>
                </SpinnerSupport>
            </div>
        </div>
    );
};

export default ItemDetail;
