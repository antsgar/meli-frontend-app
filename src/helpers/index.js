import parseCondition from "./parseCondition";
import parsePrice from "./parsePrice";

export { parseCondition, parsePrice };
