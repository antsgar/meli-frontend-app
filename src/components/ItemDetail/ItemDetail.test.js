import React from "react";
import { mount } from "enzyme";
import ItemDetail from "./ItemDetail";
import { categories, item } from "../../fixtures";
import { parseCondition, parsePrice } from "../../helpers";

let wrapper;

beforeEach(() => {
    wrapper = mount(<ItemDetail item={item} categories={categories} />);
});

describe("ItemDetail", () => {
    it("should render component", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("should display product categories", () => {
        const breadcrumb = wrapper.find("Breadcrumb");
        expect(breadcrumb.exists()).toEqual(true);
        expect(breadcrumb.find(".item")).toHaveLength(categories.length);
    });

    it("should display product information", () => {
        const { title, price, condition, quantity, description } = item;

        expect(wrapper.find(".picture").exists()).toEqual(true);
        expect(wrapper.find(".condition").text()).toEqual(
            `${parseCondition(condition)} - ${quantity} vendidos`
        );
        expect(wrapper.find(".price").text()).toEqual(parsePrice(price));
        expect(wrapper.find(".title").text()).toEqual(title);
        expect(wrapper.find(".description").text()).toEqual(description);
    });

    it("should display buy button", () => {
        const button = wrapper.find(".button");
        expect(button.exists()).toEqual(true);
        expect(button.text()).toEqual("Comprar");
    });
});
