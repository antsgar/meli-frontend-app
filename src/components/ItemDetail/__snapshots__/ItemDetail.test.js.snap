// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDetail should render component 1`] = `
<ItemDetail
  categories={
    Array [
      "Celulares y Teléfonos",
      "Celulares y Smartphones",
    ]
  }
  item={
    Object {
      "condition": "new",
      "description": "Sample description",
      "free_shipping": true,
      "id": "MLA860381954",
      "picture": "http://mla-s1-p.mlstatic.com/885508-MLA32445302787_102019-O.jpg",
      "price": Object {
        "amount": 166799,
        "currency": "ARS",
        "decimals": null,
      },
      "quantity": 4,
      "title": "iPhone 11 64 Gb Negro 4 Gb Ram",
    }
  }
>
  <div
    className="container"
    data-test="item-detail"
  >
    <div
      className="content"
    >
      <SpinnerSupport>
        <div
          className="container"
        >
          <Breadcrumb
            breadcrumbItems={
              Array [
                "Celulares y Teléfonos",
                "Celulares y Smartphones",
              ]
            }
          >
            <div
              className="container"
            >
              <span
                className="item"
                key="Celulares y Teléfonos"
              >
                Celulares y Teléfonos
              </span>
              <span
                className="item"
                key="Celulares y Smartphones"
              >
                Celulares y Smartphones
              </span>
            </div>
          </Breadcrumb>
          <div
            className="card"
          >
            <div
              className="cardContent"
            >
              <div
                className="pictureContainer"
              >
                <img
                  alt="Product picture"
                  className="picture"
                  src="http://mla-s1-p.mlstatic.com/885508-MLA32445302787_102019-O.jpg"
                />
              </div>
              <div
                className="productInfoContainer"
              >
                <div
                  className="condition"
                >
                  Nuevo
                   - 
                  4
                   vendidos
                </div>
                <div
                  className="title"
                >
                  iPhone 11 64 Gb Negro 4 Gb Ram
                </div>
                <div
                  className="price"
                >
                  ARS 166,799
                </div>
                <button
                  className="button"
                  type="button"
                >
                  Comprar
                </button>
              </div>
              <div
                className="descriptionContainer"
              >
                <div
                  className="descriptionHeader"
                >
                  Descripción
                </div>
                <div
                  className="description"
                >
                  Sample description
                </div>
              </div>
            </div>
          </div>
        </div>
      </SpinnerSupport>
    </div>
  </div>
</ItemDetail>
`;
